docs/canary_releases.md

```diff
@@ -0,0 +1,70 @@
+# Canary releases
+
+There is a need to use canary releases in order to make sure newer versions
+of the gathering conditions don't break the Insights Operator gatherer.
+
+This is how a canary release would look like:
+
+```mermaid
+sequenceDiagram
+    actor RT as Rules Team
+    actor PT as Processing Team
+    participant GC as Gathering Conditions
+    participant AI as App Interface
+    participant GCS as Gathering Conditions Service
+    participant Unleash
+    participant Storage
+    actor C as Cluster
+    actor CC as Canary Cluster
+
+    Note left of GCS: Latest version is 1.0.1
+    GCS -->> C: Get version 1.0.1
+    GCS -->> CC: Get version 1.0.1
+
+    RT ->> GC: Push a new tag 1.0.2
+    RT ->> AI: Set the canary and stable versions<br/>CANARY_VERSION = 1.0.2<br/>STABLE_VERSION = 1.0.1
+    PT ->> AI: Review and merge the PR
+    AI ->> GCS: Update prod deployment
+    
+    rect rgb(250,128,114)
+        alt canary or stable version not found
+            GCS ->> GCS: CrashLoopBackOff
+            PT ->> AI: Revert the PR
+        end
+    end
+
+    GCS -->> C: Get version 1.0.1
+    GCS -->> CC: Still get version 1.0.1
+
+    RT ->> Unleash: Increase the canary population to X%
+    GCS -->> C: Get version 1.0.1
+    rect rgb(143,188,143)
+        GCS -->> CC: Get version 1.0.2
+    end
+    
+    RT ->> Storage: Analyze the canary archives and check they are generated well
+
+    alt canary archives are wrong
+        RT ->> AI: Revert the PR
```

Comment:
I suppose PT would still have to review and merge the PR to revert the previous PR?

Reply:
As it's a revert, it can be self-serviceable from the RT. I added the PT here too just in case